manifest/list.go

```diff
@@ -0,0 +1,118 @@
+package manifest
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/containers/image/types"
+	digest "github.com/opencontainers/go-digest"
+	imgspecv1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// List is an interface for parsing, modifying lists of image manifests.
+// Callers can either use this abstract interface without understanding the
+// details of the formats, or instantiate a specific implementation (e.g.
+// manifest.OCI1Index) and access the public members directly.
+type List interface {
+	// MIMEType returns the MIME type of this particular manifest list.
+	MIMEType() string
+
+	// Instances returns a list of the manifests that this list knows of.
+	Instances() []types.BlobInfo
+
+	// Update information about the list's instances.  The length of the passed-in slice must
+	// match the length of the list of instances which the list already contains, and every field
+	// must be specified.
+	UpdateInstances([]ListUpdate) error
```

Comment:
Is this good enough?

Looking at https://github.com/containers/libpod/issues/713 , we may need the ability to add/delete/replace entries in the list, and to specify (update?) architectures.

(\*shrug\* as far as the _code_ goes, this is fine, but now that c/image is supposed to increment the major version number on API breaks… we can 1) live with this and expect an API break, 2) keep this and expect to add a separate interface method that can do more, 3) not add this public method yet, until we have a clear user that would shape the API, 4) guess at what the API should be, and add it now.

I guess this might be a good place to ask what is the primary goal of this PR right now :) )